src/org/sosy_lab/java_smt/example/NQueens.java

```diff
@@ -0,0 +1,362 @@
+
+        // solve N-Queens
+        try (ProverEnvironment prover = context.newProverEnvironment(ProverOptions.GENERATE_MODELS)) {
+            prover.push(bmgr.and(rules));
+            prover.push(bmgr.and(assignments));
+
+            boolean isUnsolvable = prover.isUnsat();
+            if (isUnsolvable) {
+                return null;
+            }
+
+            // get model and convert it
+            Boolean[][] solution = new Boolean[n][n] ;
+            try (Model model = prover.getModel()) {
+                for (int row = 0; row < n; row++) {
+                    for (int col = 0; col < n; col++) {
+
+
+                        solution[row][col] = getValue(symbols, model, row,col);
+                    }
+                }
+                return solution;
+            }
+        }
+
+
+    }
+
+
+
+
+}
+
+
+
+
+
+class NQueen extends NQueensSolver {
+
+    private final int n;
+
+    public NQueen(SolverContext context, int n) {
+        super(context);
+        this.n = n;
+    }
+
+    /**
+     * prepare symbols: one symbol for each of the N*N cells.
+     */
+    @Override
+    BooleanFormula[][] getSymbols() {
+        final BooleanFormula[][] symbols = new BooleanFormula[n][n];
+        for (int row = 0; row < n; row++) {
+            for (int col = 0; col < n; col++) {
+                symbols[row][col] = bmgr.makeVariable("q_" + row + "_" + col);
+            }
+        }
+        return symbols;
+    }
+
+  /*
+ getRules is the method used to add constraints that ensure that no two queens are in the same
+ row, column, or diagonal.
+
+ */
+
+    @Override
+    List<BooleanFormula> getRules(BooleanFormula[][] symbols, SolverContext context) {
+        List<BooleanFormula> rules = new ArrayList<>();
+
+        // Add constraints to ensure that only one queen is placed in each row
+        for (int row = 0; row < n; row++) {
+            BooleanFormula rowConstraint = bmgr.or(symbols[row]);
+            rules.add(rowConstraint);
+        }
+
+        // Add constraints to ensure that only one queen is placed in each column
+        for (int col = 0; col < n; col++) {
+            BooleanFormula[] colSymbols = new BooleanFormula[n];
+            for (int row = 0; row < n; row++) {
+                colSymbols[row] = symbols[row][col];
+            }
+            BooleanFormula colConstraint = bmgr.or(colSymbols);
+            rules.add(colConstraint);
+        }
+
+        // Add constraints to ensure that only one queen is placed in each diagonal
+        for (int row = 0; row < n; row++) {
+            for (int col = 0; col < n; col++) {
+                // Check diagonals starting from the current position to the top-left corner
+                List<BooleanFormula> diagSymbols = new ArrayList<>();
+                for (int i = row, j = col; i >= 0 && j >= 0; i--, j--) {
+                    diagSymbols.add(symbols[i][j]);
+                }
+                if (!diagSymbols.isEmpty()) {
```

Comment:
The "or"-method of BooleanFormulaManager can handle emtpy arrays/lists/collections. There is no need for the emptyness-checking here. Just apply `bmgr.or(...)` and use the result.